README.rst

```diff
@@ -0,0 +1,148 @@
+
+The ``.proto`` file starts with an optional package declaration, helps to prevent
+naming conflicts between different projects.
+
+.. code-block:: proto
+
+  syntax = "proto3";
+  package addressbook;
+
+In Chapel the generated records will be placed in a module matching the ``package``
+name. If the ``package`` name is not specified the module takes the name of the
+proto file.
+
+Next, you have to define your messages. A message is just an aggregate containing
+a set of typed fields. Many standard simple data types are available as field types,
+including int64, int32, float, double, and string.
+
+.. code-block:: proto
+
+  message Person {
+    string name = 1;
+    int32 id = 2;  // Unique ID number for this person.
+    string email = 3;
+  }
+  
+In the above example the `Person` message contains 3 fields, ``string`` type ``name``,
+``int32`` type ``id`` and ``string`` type ``email``. If a field is not set default value
+is assigned to the field by Chapel.
+
+Compiling your protocol buffers
+-------------------------------
+
+The code generator integrated with the protoc compiler toolchain
+included in the default Protocol Buffers distribution. Use the ``protoc`` command
+with the ``--chpl_out`` flag, to invoke the Chapel code generator and write the
+output ``chpl`` file to a specific location.
+location:
+
+.. code-block:: console
+
+  protoc --chpl_out=$DST_DIR $SRC_DIR/addressbook.proto
+
+This generates ``addressbook.chpl`` in your specified directory.
+
+The generated file
+------------------
+
+The generated ``addressbook.chpl`` file will contain:
+
+* A wrapper module with the name ``addressbook``.
+* A record with the name ``Person``.
+* ``name_``, ``id_``, and ``email_`` field initializers.
+* ``writeToOutputFile`` and ``parseFromInputFile`` functions for serialzation/parsing.
+
+You can import this module to a ``chpl`` file and can create an instance of ``Person``
+for populating data.
+
+.. code-block:: chpl
+
+  use addressbook;
+  use IO;
+
+  var messageObj = new Person();
+  messageObj.name = "John";
+  messageObj.id = 429496729;
+  messageObj.email = "John@a.com";
+  
+Serialzation and parsing
+------------------------
+The whole purpose of using protocol buffers is to serialize your data so that it
+can be parsed elsewhere. You can serialize your message object using the 
+``IO`` module ``writeToOutputFile`` function.
+
+.. code-block:: chpl
+  
+  var file = open("out", iomode.cw);
+  var writingChannel = file.writer();
+
+  messageObj.writeToOutputFile(writingChannel);
+  
+Parsing is also similar, each generated record has a ``parseFromInputFile``
+function. So to parse the file we have just created we can use:
+
+.. code-block:: chpl
+
+  use addressbook;
+  use IO;
+  
+  var file = open("out", iomode.r);
+  var readingChannel = file.reader();
+  
+  var messageObj = new Person();
+  messageObj.parseFromInputFile(readingChannel);
+
+
+Features
+--------
+The following features are currently supported
```

Comment:
It might be good to link from here to the protobuf definitions of these features or provide more details, something along those lines

Reply:
That sounds like a good idea. I think we can link to the corresponding definitions in the [Language guide](https://developers.google.com/protocol-buffers/docs/proto3). Also, should we mention that we currently only support `proto3` and not `proto2`?

Reply:
Yeah, I think that's a good idea (though I wouldn't say "currently", since we don't expect to support `proto2`, right?)

Reply:
`proto2` is an old version, all the features available in `proto2` are available in `proto3`(different implementations might be possible). Google has also updated the official published documentation to have generated code guides for the `proto3` version. So, I don't think there is a need to support `proto2` until and unless we face some `proto` file that we cannot handle with `proto3`. What do you think?

Reply:
I agree :)